OpenRA.Game/Actor.cs

```diff
@@ -114,6 +142,24 @@ internal Actor(World world, string name, TypeDictionary initDict)
 					// so we must ready it now, we cannot wait until all traits have finished construction.
 					if (trait is IOccupySpaceInfo)
 						OccupiesSpace = Trait<IOccupySpace>();
+
+					var provider = trait as IObservesVariables;
+					if (provider != null)
+					{
+						foreach (var variableUser in provider.GetVariableObservers())
+						{
+							allObserverNotifiers.Add(variableUser.Notifier);
+							foreach (var variable in variableUser.Variables)
+							{
+								var cs = conditionStates.GetOrAdd(variable);
+								cs.Notifiers.Add(variableUser.Notifier);
+
+								// Safeguard - in theory the conditions will be added by the time traits are notified.
+								if (!conditionCache.ContainsKey(variable))
```

Comment:
As far as I can see `conditionCache` will be empty in almost all of the cases at this point. And regardless, overwriting the value again with 0 shouldn't be a problem. That means we can remove this `if` here and just always write back `0` to save us from doing two array lookups in the worst/average case?

Reply:
Sure, the safeguard was just being overly cautious. I removed it.